packages/ilios-common/addon/modifiers/get-element.js

```diff
@@ -0,0 +1,20 @@
+import { modifier } from 'ember-modifier';
+
+/**
+  The `{{get-element}}` modifier gets a reference to an element and sends it to a method.
+
+  For example - getting the root element of a component to be used in the class.
+
+  ```handlebars
+  <div {{get-element this.doStuffWithElement}}>
+    <!-- Content -->
+  </div>
+  ```
+*/
+export default modifier(function getElement(element, [callback]) {
+  if (typeof callback === 'function') {
+    callback(element); // Pass the element to the provided callback
+  } else {
+    console.warn('get-element modifier expects a callback as the first positional argument.');
```

Comment:
Let's be harsher here. Grab `import { assert } from '@ember/debug';` and use it to ensure the callback is a function. That will throw nicely in development mode.

Reply:
I tried this approach, but couldn't get the new real test to pass for when the modifier is given an invalid/missing callback. I get a `global error` from `qunit` on the `render(hbs``)` call, and no mangling of the global window error event listeners seemed to quell it. There's no example in the codebase where we do this, so I'm currently out of ideas.

Reply:
I haven't done this one either, fun new stuff! Looks like maybe you can use `assert.Throws` from [qunit](https://qunitjs.com/api/assert/throws/).

Possibly like:

```javascript
assert.throws(await render(hbs`<div id='root-element' {{get-element}}></div>`));
```

If that doesn't work I'd be ok with skipping the second test.

Reply:
I tried my hardest, but all I get in the end is a `global failure` in `qunit` due to the throwing of an Error. I put in a `todo` for now.